service/src/test/java/uk/gov/hmcts/reform/fpl/validators/HasEPOGroundsValidatorTest.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.hmcts.reform.fpl.validators;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.enums.OrderType;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.GroundsForEPO;
+import uk.gov.hmcts.reform.fpl.model.Orders;
+
+import javax.validation.ConstraintValidatorContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(SpringExtension.class)
+class HasEPOGroundsValidatorTest {
+    private HasEPOGroundsValidator validator = new HasEPOGroundsValidator();
+
+    @Mock
+    private ConstraintValidatorContext constraintValidatorContext;
+
+    @Mock
+    private ConstraintValidatorContext.ConstraintViolationBuilder constraintViolationBuilder;
+
+    @Mock
+    private ConstraintValidatorContext.ConstraintViolationBuilder.NodeBuilderCustomizableContext
+        nodeBuilderCustomizableContext;
+
+    @BeforeEach
+    private void prepareMocks() {
+        given(constraintValidatorContext.getDefaultConstraintMessageTemplate()).willReturn("");
+        given(constraintValidatorContext.buildConstraintViolationWithTemplate(Mockito.anyString()))
+            .willReturn(constraintViolationBuilder);
+        given(constraintViolationBuilder.addPropertyNode(Mockito.anyString()))
+            .willReturn(nodeBuilderCustomizableContext);
+    }
```

Comment:
```suggestion
            @Mock(answer = Answers.RETURNS_DEEP_STUBS)
	    private ConstraintValidatorContext constraintValidatorContext;
 
```
You don't need then to mock anything else. Should you verify if the appropriate node was added to the context?

Reply:
I would deprecate my comment with the suggestion below - just use validation.buildDefaultValidationFactory() and use real validators. Test smaller classes to see how your validators behave.